### DISCUSSION TITLE
calcGrains() function in Mtex 5.5.2

### DISCUSSION THREAD

**Discussion by AnweshaK91** [mtex-toolbox/mtex#752] [Ask Anything]
Hi Ralf,

I am using Mtex 5.5.2. After importing the EBSD data file in .ang format using the import_wizard I am getting error while running the calcGrains function. Below is the script I am have used to calculate the grains:
```matlab
[grains, ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',10*degree);
```
I am getting the following error:
```
Error using  ~= 
Matrix dimensions must agree.

Error in grainBoundary/calcTriplePoints (line 25)
iV = reshape(iV(iV ~= find(itP).').',3,[]).';

Error in grainBoundary (line 124)
      gB.triplePoints = gB.calcTriplePoints(V,grainsPhaseId);

Error in grain2d (line 189)
      grains.boundary = grainBoundary(V,F,I_FDext,ebsd,grains.phaseId);

Error in EBSD/calcGrains (line 60)
grains = grain2d(ebsd,V,F,I_DG,I_FD,A_Db,varargin{:});

Error in ImageAnalysis (line 36)
[grains, ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',10*degree);
```
The same error comes if I use 'threshold' option instead of 'angle'.
However this problem does not come in the older versions of Mtex like 5.3.1. I am using Matlab 2016.
Could you please suggest as to how I can overcome this problem?

With Regards,

Anwesha

**Comment by lfgmorales** [comment]
Hi

Do you get this error if you removed (‚indexed‘) from the calcGrains line?

Cheers


Luiz

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 6 Feb 2021, at 06:00, AnweshaK91 <notifications@github.com> wrote:

﻿

Hi Ralf,

I am using Mtex 5.5.2. After importing the EBSD data file in .ang format using the import_wizard I am getting error while running the calcGrains function. Below is the script I am have used to calculate the grains:

[grains, ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',10*degree);

I am getting the following error:
Error using ~=
Matrix dimensions must agree.

Error in grainBoundary/calcTriplePoints (line 25)
iV = reshape(iV(iV ~= find(itP).').',3,[]).';

Error in grainBoundary (line 124)
gB.triplePoints = gB.calcTriplePoints(V,grainsPhaseId);

Error in grain2d (line 189)
grains.boundary = grainBoundary(V,F,I_FDext,ebsd,grains.phaseId);

Error in EBSD/calcGrains (line 60)
grains = grain2d(ebsd,V,F,I_DG,I_FD,A_Db,varargin{:});

Error in ImageAnalysis (line 36)
[grains, ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',10*degree);

The same error comes if I use 'threshold' option instead of 'angle'.
However this problem does not come in the older versions of Mtex like 5.3.1. I am using Matlab 2016.
Could you please suggest as to how I can overcome this problem?

With Regards,

Anwesha

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/752>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5INY5FKYYFMJMX54ATT3S5TEHRANCNFSM4XF4LOKA>.

[ { "@context": "http://schema.org", "@type": "EmailMessage", "potentialAction": { "@type": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/752", "url": "https://github.com/mtex-toolbox/mtex/discussions/752", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { "@type": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by JianhuaLiu** [reply]
I get the same problem when I use the command 'calcGrains' , I am using Matlab 2016a in the versions of Mtex 5.4 and the newest 5.6, both don't work, then I used Matlab 2021a, now it works well.

**Reply by ralfHielscher** [reply]
Could you please check out MTEX 5.6.1?

**Reply by JianhuaLiu** [reply]
I check out MTEX 5.6.1, but I can't even install it, I tried  by using the command 'startup_mtex' and runing the install_mtex.m script directly, both failed, maybe there is something wrong with it, Now I reinstall MTEX 5.6.0 back.

**Comment by AnweshaK91** [comment]
Hi Luiz,

Yes I get this error even when I remove 'indexed' from the calcGrains() function and just use ebsd object name in the function.

With regards,

Anwesha
________________________________

**Comment by lfgmorales** [comment]
Hi Anwesha


Can you update to MTEX 5.6 and see what happens, I cannot reproduce your error, so it might be (i) your EBSD file or (ii) your matlab, we know MTEX should work fine from Matlab 2016b


do you get any error when using import_wizard?



Cheers



Luiz





Luiz F. G. Morales
ScopeM & Structural Geology and Tectonics Group
ETH Zürich
________________________________

**Comment by AnweshaK91** [comment]
Sure Luiz I will try with version 5.6. The calcGrains function is working fine in version 5.3.1, I have tried in it. I do not get any error while using the import_wizard. The import_wizard,  first lists the non indexed phase as -1 and then the indexed phase a phase 0 in both the versions.

Anwesha

________________________________

**Comment by ralfHielscher** [comment]
What kind of error message do you get?
